### PR TITLE
[CIS-540] Save channel data from message payload

### PR DESCRIPTION
**Pull request by VojtaStavik**
Firstly - the way we decoded message endpoint responses were always wrong and never worked 🤷‍♂️ :))

Secondly - We found out that our backed _sometimes_ includes channel payload directly into message endpoint response. This is basically what solves issue described in CIS-320. This PR makes it possible to parse and save channel-related data to DB, too.

This is a preparation PR for adding push notification support to LLC. Thanks to this, we will have to make just one API call, instead of two.

---

This PR also closes CIS-320.

### CODE REVIEW

File: Sources_v3/StreamChat/APIClient/Endpoints/Payloads/MessagePayloads.swift [hunk 1]
```diff
@@ -25,12 +25,16 @@ enum MessagePayloadsCodingKeys: String, CodingKey {
     case ownReactions = "own_reactions"
     case reactionScores = "reaction_scores"
     case isSilent = "silent"
+    case channel
     //        case i18n
 }
 
-/// An object describing the wrapped incoming message JSON payload.
-struct WrappedMessagePayload<ExtraData: ExtraDataTypes>: Decodable {
-    let message: MessagePayload<ExtraData>
```
Comment (b-onc) [inline Sources_v3/StreamChat/APIClient/Endpoints/Payloads/MessagePayloads.swift:33]: I liked `WrappedMessage` more than `MessagePayload.Box` :( Any particular reason to change it?

**Review by b-onc — APPROVED** [review]
LGTM

File: Sources_v3/StreamChat/APIClient/Endpoints/Payloads/MessagePayloads.swift (hunk 1 above)
Comment (VojtaStavik) [reply Sources_v3/StreamChat/APIClient/Endpoints/Payloads/MessagePayloads.swift:33]: @b-onc I didn't like the name collision with optional wrapped type + I kinda like having this a nested type. Happy to hear your suggestions regarding the name.

Comment (ManWithBear) [reply Sources_v3/StreamChat/APIClient/Endpoints/Payloads/MessagePayloads.swift:33]: I'm a `Box` guy

File: Sources_v3/StreamChat/APIClient/Endpoints/Payloads/MessagePayloads.swift [hunk 2]
```diff
@@ -59,6 +60,10 @@ struct MessagePayload<ExtraData: ExtraDataTypes>: Decodable {
     let reactionScores: [MessageReactionType: Int]
     let attachments: [AttachmentPayload<ExtraData.Attachment>]
     let isSilent: Bool
+
+    /// Only message payload from `getMessage` endpoint contains channel data. It's a convenience workaround for having to
+    /// mane an extra call do get channel details.
```
Comment (ManWithBear) [inline Sources_v3/StreamChat/APIClient/Endpoints/Payloads/MessagePayloads.swift:65]: nit: `make`?

**Review by ManWithBear — APPROVED** [review]

File: Sources_v3/StreamChat/APIClient/Endpoints/Payloads/MessagePayloads.swift (hunk 1 above)
Comment (ManWithBear) [reply Sources_v3/StreamChat/APIClient/Endpoints/Payloads/MessagePayloads.swift:33]: On other hand from usage point of view I would expect `ed` suffix:
```
let endpoint: Endpoint<MessagePayload<DefaultExtraData>.Boxed>
let endpoint: Endpoint<MessagePayload<DefaultExtraData>.Wrapped>
```

File: Sources_v3/StreamChat/APIClient/Endpoints/Payloads/MessagePayloads.swift (hunk 2 above)
Comment (VojtaStavik) [reply Sources_v3/StreamChat/APIClient/Endpoints/Payloads/MessagePayloads.swift]: fixed in f639bfb

File: Sources_v3/StreamChat/APIClient/Endpoints/Payloads/MessagePayloads.swift (hunk 1 above)
Comment (VojtaStavik) [reply Sources_v3/StreamChat/APIClient/Endpoints/Payloads/MessagePayloads.swift:33]: I like the `Boxed` idea. Fixed in 2da2168

**Pull request merged**